src/core/p5.Renderer.js

```diff
@@ -35,69 +40,107 @@ p5.Renderer = class Renderer extends p5.Element {
       this._styles = []; // non-main elt styles stored in p5.Renderer
     }
 
+    // Renderer state machine
+    this.states = {
+      doStroke: true,
+      strokeSet: false,
+      doFill: true,
+      fillSet: false,
+      tint: null,
+      imageMode: constants.CORNER,
+      rectMode: constants.CORNER,
+      ellipseMode: constants.CENTER,
+      textFont: 'sans-serif',
+      textLeading: 15,
+      leadingSet: false,
+      textSize: 12,
+      textAlign: constants.LEFT,
+      textBaseline: constants.BASELINE,
+      textStyle: constants.NORMAL,
+      textWrap: constants.WORD
+    };
+    this.pushPopStack = [];
+
+
+
     this._clipping = false;
     this._clipInvert = false;
 
-    this._textSize = 12;
-    this._textLeading = 15;
-    this._textFont = 'sans-serif';
-    this._textStyle = constants.NORMAL;
-    this._textAscent = null;
-    this._textDescent = null;
-    this._textAlign = constants.LEFT;
-    this._textBaseline = constants.BASELINE;
-    this._textWrap = constants.WORD;
-
-    this._rectMode = constants.CORNER;
-    this._ellipseMode = constants.CENTER;
+    // this._textSize = 12;
+    // this._textLeading = 15;
+    // this._textFont = 'sans-serif';
+    // this._textStyle = constants.NORMAL;
+    // this._textAscent = null;
+    // this._textDescent = null;
+    // this._textAlign = constants.LEFT;
+    // this._textBaseline = constants.BASELINE;
+    // this._textWrap = constants.WORD;
+
+    // this._rectMode = constants.CORNER;
+    // this._ellipseMode = constants.CENTER;
     this._curveTightness = 0;
-    this._imageMode = constants.CORNER;
+    // this._imageMode = constants.CORNER;
 
-    this._tint = null;
-    this._doStroke = true;
-    this._doFill = true;
-    this._strokeSet = false;
-    this._fillSet = false;
-    this._leadingSet = false;
+    // this._tint = null;
+    // this._doStroke = true;
+    // this._doFill = true;
+    // this._strokeSet = false;
+    // this._fillSet = false;
+    // this._leadingSet = false;
 
     this._pushPopDepth = 0;
   }
 
+  id(id) {
+    if (typeof id === 'undefined') {
+      return this.elt.id;
+    }
+
+    this.elt.id = id;
+    this.width = this.elt.offsetWidth;
+    this.height = this.elt.offsetHeight;
+    return this;
+  }
```

Comment:
This will need to be refactored, possibly into individual renderer. It is copied from p5.Element because accessibility features are expecting it.